source/Concepts/Advanced/About-Internal-Interfaces.rst

```diff
@@ -191,3 +191,58 @@ These are mainly used for error handling, commandline argument parsing, and logg
 The ``rcutils`` |API| and implementation are located in the `ros2/rcutils <https://github.com/ros2/rcutils>`_ repository on |GitHub|_ which contains the interface as C headers.
 
 For a complete definition of the ``rcutils`` |API|, see `the rcutils docs <https://docs.ros.org/en/{DISTRO}/p/rcutils/>`_.
+
+Automatic code generation using template files
+==============================================
+
+As mentioned above, in order to deal with different DDS implementations, code has to be generated. Code generation is handled by a clever usage
+of CMake and Ament. To simplify the explanation of this mechanism let's separate ROS2 projects that generate or need code to be generated in two types.
+
+-  ``Generator`` projects, which are the ones that have the templates and logic to generate code.
+-  ``Client`` projects, which are the ones that need code to be generated, they have the parameters to generate code.
```

Comment:
Client as a keyword collides with the "ros client library" verbiage. I might suggest an alternative to `client` to differentiate from `rcl`. I would suggest potentially `interface` projects.